third-party/x-crypto/argon2id/argon2id.go

```
@@ -0,0 +1,176 @@
+// compares in constant time. A malformed phc returns InvalidPasswordHash; a
+// genuine mismatch is (false, nil).
+func (argon2idPW) Verify(password []byte, phc string) (ok bool, err error) {
+	//: decode the stored parameters; a parse failure is data corruption.
+	memCost, timeCost, threads, salt, want, decoded := decodePHC(phc)
+	//: reject an unparseable stored hash before any work.
+	if !decoded {
+		//: server-side corruption, not a password mismatch.
+		return false, corecrypto.InvalidPasswordHash
+	}
+	//: recompute with the SAME params + output length as the stored digest.
+	got := argon2.IDKey(password, salt, timeCost, memCost, threads, uint32(len(want)))
```

golangci-lint (2.12.2)

[high] G115: integer overflow conversion int -> uint32

(gosec)